maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultArtifactDescriptorReader.java

```diff
@@ -243,9 +243,10 @@ private Model loadPom( RepositorySystemSession session, ArtifactDescriptorReques
             }
             catch ( ArtifactResolutionException e )
             {
-                if ( e.getCause() instanceof ArtifactNotFoundException )
+                Throwable cause = e.getCause();
+                if ( cause instanceof ArtifactTransferException )
                 {
-                    missingDescriptor( session, trace, a, (Exception) e.getCause() );
+                    missingDescriptor( session, trace, a, (ArtifactTransferException) cause );
```

Comment:
`ArtifactNotFoundException`  is instance of `ArtifactTransferException`. Previous condition to check both did not make sense.
Clarified cast to be more specific class.